charts/signoz/templates/otel-collector-metrics/deployment.yaml

```
@@ -147,6 +150,9 @@ spec:
         - name: otel-collector-metrics-config-vol
           configMap:
             name: {{ include "otelCollectorMetrics.fullname" . }}
+{{- if .Values.otelCollectorMetrics.extraVolumes }}
+{{ toYaml .Values.otelCollectorMetrics.extraVolumes | indent 8 }}
+{{- end }}
```

same as above

---

The problem with that is if you ident it the output look like this:
```
      volumes:
        - name: otel-collector-metrics-config-vol
          configMap:
            name: test-signoz-otel-collector-metrics
                - name: test-pvc
          persistentVolumeClaim:
            claimName: pvc-xxxxxxxxxx
```
And give this error:
```
Error: YAML parse error on signoz/templates/otel-collector-metrics/deployment.yaml: error converting YAML to JSON: yaml: line 150: mapping values are not allowed in this context
helm.go:84: [debug] error converting YAML to JSON: yaml: line 150: mapping values are not allowed in this context
```
Except you have another solution, no indentation work

---

You could try `nindent` instead.

Can you try the following?

```suggestion
        {{- if .Values.otelCollectorMetrics.extraVolumes }}
          {{- toYaml .Values.otelCollectorMetrics.extraVolumes | nindent 8 }}
        {{- end }}
```

---

Oh that's work ! Thx for that, Done in: https://github.com/SigNoz/charts/pull/270/commits/25dfe02fbfa9bbce3ae37f6e9814224858b2dfcf and https://github.com/SigNoz/charts/pull/270/commits/f131c9ed76a8f77f621640fe13e63edf9d2e909d